Install nginx on all servers.

smarnach opened the pull request:
This PR installs nginx on all servers, and enables requesting certificates with Certbot by default.  We need nginx on all servers to reverse-proxy the Prometheus exporters.

Note that I removed the cron job to renew certificates, since it is supplied by the apt packages now.  We might end up with two cron jobs on some servers, since Ansible does not remove the cron jobs we remove from the config, but that shouldn't be a problem – the server will check for new certs four times a day insted of just two times.

smarnach committed:
Install nginx on all servers.

smarnach committed:
Remove cert renewal cron job.

The cron job is provided by the certbot apt package now, so we don't need to
take care of it anymore.

smarnach committed:
Set up global SSL settings for nginx.

UmanShahzad commented on tasks/nginx.yml:
```diff
@@ -0,0 +1,79 @@
+        path: /var/www/certbot
+        state: directory
+
+    - name: generate SSL certificate with certbot
```
Although this works, the job that the apt package installs is this:

```bash
# /etc/cron.d/certbot: crontab entries for the certbot package
#
# Upstream recommends attempting renewal twice a day
#
# Eventually, this will be an opportunity to validate certificates
# haven't been revoked, etc.  Renewal will only occur if expiration
# is within 30 days.
SHELL=/bin/sh
PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin

0 */12 * * * root test -x /usr/bin/certbot -a \! -d /run/systemd/system && perl -e 'sleep int(rand(3600))' && certbot -q renew
```

I tried running that command with a dry run and it fails because:

```
Attempting to renew cert (rabbitmq-stage.opencraft.hosting) from /etc/letsencrypt/renewal/rabbitmq-stage.opencraft.hosting.conf produced an unexpected error: Problem binding to port 80: Could not bind to IPv4 or IPv6.. Skipping.
All renewal attempts failed. The following certs could not be renewed:
  /etc/letsencrypt/live/rabbitmq-stage.opencraft.hosting/fullchain.pem (failure)
```

So unless I did something wrong, I think we'd need to delete this default job and add one ourselves that uses the webroot plugin. The command that I used that worked in a dry run was the same that you used below. Using `renew` rather than `certonly` gives:

```
Currently, the renew verb is capable of either renewing all installed certificates that are due to be renewed or renewing a single certificate specified by its name. If you would like to renew specific certificates by their domains, use the certonly command instead. The renew verb may provide other options for selecting certificates to renew in the future.
```

so the command below is perfect in the cronjob.

smarnach replied:
@UmanShahzad Certbot will reuse whatever auth plugin it used when the certificate was originally requested.  I may have to manually fix up the renewal configurations of existing certificates when I deploy this.

These servers could be affected:
* `chat.opencraft.com` – I already fixed that one manually.
* `rabbitmq*.opencraft.hosting` – these four servers need manual fixing.  I already fixed rabbitmq-stage in response to this comment, so `certbot renew --dry-run` now works.
* The accounting server should be fine without manual fixing.  It should have the old pre-hook and post-hook in its renewal config, so it should work correctly.  I can't verify this, though, so you may have to fix it up manually as well.

To fix this manually, we need to edit the configuration file in `/etc/letsencrypt/renewal/` to include these settings:
```
authenticator = webroot
webroot_path = /var/www/certbot,
[[webroot_map]]
chat.opencraft.com = /var/www/certbot
```
The last line needs to be adapted, of course.  I don't know whether the trailing comma for the `webroot_path` setting is required, but it's there in the auto-generated config, so I just left it in.

UmanShahzad reviewed (changes requested):
This is looking perfect except 1 thing -- I noticed the default renewal job from certbot isn't working as intended -- it's not using the webroot plugin and then it collides with nginx. Or maybe I did something wrong?

smarnach commented:
@UmanShahzad I think I've addressed your comment, so I will merge this now, since I have a follow-up PR which is based on this one.  I can address any further comments in the follow-up.
